docs/architecture-decisions/2023-06-06_frontend-templates.md

```diff
@@ -13,23 +13,54 @@ AlgoKit v2 aims provide an end-to-end development and deployment experience that
+
+### Option 1: Monorepos
+
+This option suggests that the frontend and backend templates should be bundled together in a single repository. This approach aims to simplify the maintenance of the templates by reducing the number of repositories that need to be maintained. Both official and community templates should be build on top of this approach where frontend is always tailored for specific needs of the smart contracts that are being used.
 
-Expanding AlgoKit templating capabilities is a non-trivial task that requires careful consideration of the trade-offs between the different approaches. The following section outlines the different options that were considered and the trade-offs associated with each of them.
+Drawbacks:
 
-> Refer to git history to see old revisions of the document. After further discussions with the team we are now aligned on the separate repositories approach. Hence the sections below are updated to reflect the new approach.
+- **Lack of flexibility**: The monorepo approach does not allow developers to choose their preferred frontend technology stack. This is because the frontend is tightly coupled with the backend and the two cannot be separated. This is a major drawback as it limits the flexibility of the templates and does not allow developers to choose their preferred frontend technology stack.
+- **Increased maintenance overhead**: The monorepo approach increases the maintenance overhead as it requires the templates to be updated together.
+
+### Option 2: Separate Repositories
+
+This option suggests that the frontend and backend templates should be maintained in separate repositories. This approach aims to provide developers with the flexibility to choose their preferred frontend technology stack. It also allows the templates to be maintained independently, thereby reducing the maintenance overhead.
+
+After several review rounds team reached a conclusion that this approach is not ideal for the following reasons:
+
+Drawbacks:
+
+- **Compatibility overhead**: The separate repositories approach increases the maintenance overhead as it requires the templates to be updated independently. This can lead to issues with compatibility between the templates and the typed clients.
+- **Lack of guidance**: The separate repositories approach does not provide developers with a clear process for linking the frontend templates with the typed clients generated by the backend templates. This can lead to confusion and issues with the integration of the templates.
+
+After several review rounds team reached a conclusion to expand on this option given better alignment with AlgoKit design principles. Hence, the following proposal is further exploration of ideas based on this approach and propose solution to mitigate the potential drawbacks and risks outlined.
```

Comment:
Minor: You can probably drop this paragraph and have a section towards the end that describes the selected option based on all of the options.

Reply:
I agree this would be clearer, as I was a bit confused by these two conflicting statements:

>After several review rounds team reached a conclusion that this approach is not ideal for the following reasons:

>After several review rounds team reached a conclusion to expand on this option given better alignment with AlgoKit design principles